CMakeLists.txt

```diff
@@ -1,9 +1,21 @@
 # Compatibility
-cmake_minimum_required(VERSION 2.6)
-
-if(WIN32)
-    # Windows XP compatibility. Must be set before everything else.
-    set(CMAKE_GENERATOR_TOOLSET "v120_xp")
+cmake_minimum_required(VERSION 3.7)
+
+# Windows XP compatible platform toolset. Must be set before project(),
+# otherwise change of CMAKE_*_TOOLSET will take no effect.
+# We get VS version from the generator name because neither MSVC* nor other
+# variables that describe the compiler aren't available before project().
+if ("${CMAKE_GENERATOR}" MATCHES "Visual Studio ([0-9]+)")
```

Comment:
Can you put this *and* `cmake_minimum_required` inside `if (WIN32)`,
and leave version at 2.6 for other platforms?